docs/cody/proxy-setup.mdx

```diff
@@ -0,0 +1,112 @@
+
+# Proxy Setup Guide
+
+## Introduction
+This guide will walk you through setting up HTTP, HTTPS, and SOCKS proxies in Visual Studio Code (VS Code) and JetBrains IDEs. It also includes instructions for handling self-signed certificates on macOS and Windows.
+
+## Proxy Setup for VS Code
+
+### Accessing the Proxy Settings
+1. Open VS Code.
+2. Navigate to the settings window of VS Code.
+3. In the search bar at the top of the Settings pane, type `proxy`.
+4. Click on `Edit in settings.json` to open the `settings.json` file.
+
+### Modifying `settings.json` for Proxy Configuration
+Add the following configuration fields in the `settings.json` file to set up your proxy:
+
+```json
+{
+  "http.proxy": "http://proxy_ip:port", // Replace with your proxy IP and port
+  "http.proxySupport": "on",
+  "http.proxyStrictSSL": true,
+  "http.systemCertificates": true,
+  "http.experimental.systemCertificatesV2": true
+}
```

Comment:
@arafatkatze Do I need both?

Reply:
Yes you do need both. This was a dealbreaker when testing with self signed certificates